### PR TITLE
CC-16985: Unzer Glue support integration

### PR DESCRIPTION
**Pull request by volodymyr-bezuhlyi**
#### Overview

- Ticket: https://spryker.atlassian.net/browse/CC-16985

###### Optional

Reference all core PRs for auto-link back-references.

Core PRs
- https://github.com/spryker/suite-nonsplit/pull/6882

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -10,6 +10,7 @@
         "spryker/calculation": "^4.0.0",
         "spryker/cart-extension": "^4.1.0",
         "spryker/checkout-extension": "^1.4.0",
+        "spryker/checkout-rest-api-extension": "^1.5",
```
Comment (AsonUnique) [inline composer.json]:
```suggestion
        "spryker/checkout-rest-api-extension": "^1.5.0",
```

File: src/SprykerEco/Zed/Unzer/Communication/Plugin/CheckoutRestApi/UnzerCheckoutDataExpanderPlugin.php [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+
+use Generated\Shared\Transfer\RestCheckoutDataTransfer;
+use Generated\Shared\Transfer\RestCheckoutRequestAttributesTransfer;
+use Spryker\Zed\CheckoutRestApiExtension\Dependency\Plugin\CheckoutDataExpanderPluginInterface;
+use Spryker\Zed\Kernel\Communication\AbstractPlugin;
+
+/**
+ * @method \SprykerEco\Zed\Unzer\Business\UnzerFacadeInterface getFacade()
+ */
+class UnzerCheckoutDataExpanderPlugin  extends AbstractPlugin implements CheckoutDataExpanderPluginInterface
```
Comment (AsonUnique) [inline src/SprykerEco/Zed/Unzer/Communication/Plugin/CheckoutRestApi/UnzerCheckoutDataExpanderPlugin.php]:
```suggestion
class UnzerCheckoutDataExpanderPlugin extends AbstractPlugin implements CheckoutDataExpanderPluginInterface
```

**Review by AsonUnique — APPROVED** [review]

File: src/SprykerEco/Shared/Unzer/Transfer/unzer.transfer.xml [hunk 3]
```diff
@@ -793,4 +793,11 @@
         <property name="idSalesShipment" type="int"/>
         <property name="chargeId" type="string"/>
     </transfer>
+
+    <transfer name="RestCheckoutData">
+        <property name="quote" type="Quote"/>
+    </transfer>
+
+    <transfer name="Quote">
+    </transfer>
```
Comment (ilyakubanov) [inline src/SprykerEco/Shared/Unzer/Transfer/unzer.transfer.xml]: already defined

File: src/SprykerEco/Zed/Unzer/Business/Quote/UnzerQuoteExpander.php [hunk 4]
```diff
@@ -99,7 +100,7 @@ public function expand(QuoteTransfer $quoteTransfer): QuoteTransfer
      */
     protected function addUnzerPayment(QuoteTransfer $quoteTransfer): QuoteTransfer
     {
-        $paymentResourceTransfer = $quoteTransfer->getPaymentOrFail()->getUnzerPaymentOrFail()->getPaymentResource();
+        $paymentResourceTransfer = $this->getUnzerPaymentResource($quoteTransfer);
```
Comment (ilyakubanov) [inline src/SprykerEco/Zed/Unzer/Business/Quote/UnzerQuoteExpander.php]:
```suggestion
        $unzerPaymentResourceTransfer = $this->getUnzerPaymentResource($quoteTransfer);
```

File: src/SprykerEco/Zed/Unzer/Communication/Plugin/CheckoutRestApi/UnzerCheckoutDataExpanderPlugin.php [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+/**
+ * @method \SprykerEco\Zed\Unzer\Business\UnzerFacadeInterface getFacade()
+ * @method \SprykerEco\Zed\Unzer\UnzerConfig getConfig()
+ * @method \SprykerEco\Zed\Unzer\Communication\UnzerCommunicationFactory getFactory()
+ */
+class UnzerCheckoutDataExpanderPlugin extends AbstractPlugin implements CheckoutDataExpanderPluginInterface
+{
+    /**
+     * {@inheritDoc}
+     * - Expands `RestCheckoutDataTransfer` quote with `UnzerCredentialsTransfer`.
```
Comment (ilyakubanov) [inline src/SprykerEco/Zed/Unzer/Communication/Plugin/CheckoutRestApi/UnzerCheckoutDataExpanderPlugin.php]: please expand the specification (use facade method specification as a reference)

**Review by ilyakubanov — COMMENTED** [review]
- please create dev epic

File: src/SprykerEco/Zed/Unzer/Communication/Plugin/CheckoutRestApi/UnzerCheckoutDataExpanderPlugin.php [hunk 6]
```diff
@@ -0,0 +1,46 @@
...
+ * @method \SprykerEco\Zed\Unzer\UnzerConfig getConfig()
+ * @method \SprykerEco\Zed\Unzer\Communication\UnzerCommunicationFactory getFactory()
+ */
+class UnzerCheckoutDataExpanderPlugin extends AbstractPlugin implements CheckoutDataExpanderPluginInterface
+{
+    /**
+     * {@inheritDoc}
+     * - Requires `RestCheckoutDataTransfer.Quote` transfer property to be set.
+     * - Expands `RestCheckoutDataTransfer.QuoteTransfer` with `UnzerCredentialsTransfer` according to added items.
+     * - Does nothing if quote has no items.
```
Comment (ilyakubanov) [inline src/SprykerEco/Zed/Unzer/Communication/Plugin/CheckoutRestApi/UnzerCheckoutDataExpanderPlugin.php]:
```suggestion
     * - Requires `RestCheckoutDataTransfer.quote.store.name` transfer property to be set.
     * - Expands `RestCheckoutDataTransfer.quote` transfer property with `UnzerCredentialsTransfer` according to added items.
     * - Does nothing if quote has no items.
```

**Review by ilyakubanov — APPROVED** [review]

**Review by kkicheglovspryker — APPROVED** [review]

File: src/SprykerEco/Zed/Unzer/Communication/Plugin/CheckoutRestApi/UnzerCheckoutDataExpanderPlugin.php [hunk 7]
```diff
@@ -0,0 +1,46 @@
...
+ * @method \SprykerEco\Zed\Unzer\Communication\UnzerCommunicationFactory getFactory()
+ */
+class UnzerCheckoutDataExpanderPlugin extends AbstractPlugin implements CheckoutDataExpanderPluginInterface
+{
+    /**
+     * {@inheritDoc}
+     * - Requires `RestCheckoutDataTransfer.quote.store.name` transfer property to be set.
+     * - Expands `RestCheckoutDataTransfer.quote` transfer property with `UnzerCredentialsTransfer` according to added items.
+     * - Does nothing if quote has no items.
+     *
```
Comment (DmitryLymarenko) [inline src/SprykerEco/Zed/Unzer/Communication/Plugin/CheckoutRestApi/UnzerCheckoutDataExpanderPlugin.php:27]: we need to mention this action at specification

`return $restCheckoutDataTransfer->setQuote($quoteTransfer);`

**Review by DmitryLymarenko — APPROVED** [review]

File: src/SprykerEco/Shared/Unzer/Transfer/unzer.transfer.xml (hunk 3 above)
Comment (michbeck) [reply src/SprykerEco/Shared/Unzer/Transfer/unzer.transfer.xml]: Resolved in consultation with Ilya Kubanov.

File: src/SprykerEco/Zed/Unzer/Business/Quote/UnzerQuoteExpander.php (hunk 4 above)
Comment (michbeck) [reply src/SprykerEco/Zed/Unzer/Business/Quote/UnzerQuoteExpander.php]: Resolved in consultation with Ilya Kubanov.

File: src/SprykerEco/Zed/Unzer/Communication/Plugin/CheckoutRestApi/UnzerCheckoutDataExpanderPlugin.php (hunk 5 above)
Comment (michbeck) [reply src/SprykerEco/Zed/Unzer/Communication/Plugin/CheckoutRestApi/UnzerCheckoutDataExpanderPlugin.php]: Resolved in consultation with Ilya Kubanov.

**Review by DmitryLymarenko — APPROVED** [review]

**Pull request merged**